functions/cloud_functions/userCreated.js

```diff
@@ -0,0 +1,53 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin')
+
+
+exports.userCreated = functions.auth.user().onCreate(async user => {
+
+    const db = admin.firestore()
+
+    const newUser = {
+        firstName : "",
+        lastName : "",
+        middleName : "",
+        phone : "",
+        kinFirstName : "",
+        kinLastName : "",
+        kinMiddleName : "",
+        kinPhone : "",
+        bankName : "",
+        accountName : "",
+        accountNumber : ""
+    }
+
+    const privateInfo = {
+        uid : user.uid,
+        createdAt : new Date().getTime() + (new Date().getTimezoneOffset() * 60000),
+        email : user.email,
+        wallet : {
+            pending : 0,
+            available : 0
+        },
+        walletTransactions : [],
+        role : "user",
+        ROI : 0,
+        assignedTrader : "",
+        approved : false,
+        affiliate : ""
+    }
+
+
+    try {
+
+        const promises = []
+
+        promises.push(await db.doc(`users/${user.uid}`).set(newUser))
```

Comment:
Wht are you creating the user data with an empty data?

Reply:
because the register form form comes with other data apart from the email and password. Firebase createUser function only works with email and password...so the other data will be lost. Instead I save those data in another document in the database...Those data will be merged once the admin approves the account.

Reply:
Yh, I get the firebase part but the `newUser` object doesn't really have the any data attached to it. So it should be

Reply:
Not sure I get ya.... there is a `const newUser` up in the file...